src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -3506,25 +3530,19 @@ namespace winrt::Microsoft::Terminal::Control::implementation
                 L"SearchBoxResultAnnouncement" /* unique name for this group of notifications */);
         }
 
-        // Manually send a scrollbar update, now, on the UI thread. We're
-        // already UI-driven, so that's okay. We're not really changing the
-        // scrollbar, but we do want to update the position of any marks. The
-        // Core might send a scrollbar updated event too, but if the first
-        // search hit is in the visible viewport, then the pips won't display
-        // until the user first scrolls.
-        auto scrollBar = ScrollBar();
-        ScrollBarUpdate update{
-            .newValue = scrollBar.Value(),
-            .newMaximum = scrollBar.Maximum(),
-            .newMinimum = scrollBar.Minimum(),
-            .newViewportSize = scrollBar.ViewportSize(),
-        };
-        _throttledUpdateScrollbar(update);
+        _UpdateSearchScrollMarks();

```

Comment:
...... ah yea here it is. We always send the update if we got an explicit change to the search marks, which we will when the buffer is circling.